src/debugger/client.js

```diff
@@ -0,0 +1,57 @@
+const http = require('http');
+const createRPC = require('multiplex-rpc');
+const wsock = require('websocket-stream');
+const eos = require('end-of-stream');
+const { decode } = require('./snapshot');
+
+const methods = ['step', 'next', 'previous'];
+
+class Client {
```

Comment:
To keep with more style of the rest of the codebase I'd prefer if this was written in terms of a closure rather than a class:

```javascript
const createClient = (ws, rpc, plex) => ({
  close: () => {...},
  step: () => {...}
});
```